### ISSUE TITLE
ADLnet constants endpoint 404 error

### ISSUE THREAD

**Issue by T1ckL35** [adlnet/SCORM-to-xAPI-Wrapper#4]
Hi,
I'm just looking at implementing this (or using it as a basis for a custom implementation) and after the initial setup am getting a 404 for any requests to the adlnet endpoints:
Anything to http://adlnet.gov/xapi and sub locations e.g, http://adlnet.gov/xapi/profile/scorm/activity-state all fail with an nginx 404.

Do you know if this address has moved and to where?
I can see that the SCORMToXAPIFunctions.js code setting the constants object is where it is falling down for me.

Any help would be appreciated.
Simon

**Comment by stephenlane** [comment]
Can you post the line that it is failing on and whatever error you are getting

**Comment by T1ckL35** [comment]
Hi,

I've set the wrapper up with Learning Locker and hooked it into our own LMS and am getting the following in the LRS log which seems to kill things:

2015-05-18 13:26:14] local.ERROR: exception 'Locker\Helpers\Exceptions\NotFound' with message 'Could not find state with id `http://adlnet.gov/xapi/profile/scorm/attempt-state`' in ****/learninglocker/app/controllers/xapi/DocumentController.php:224

Maybe it is something to do with the implementation with Learning Locker or just my misunderstanding of how it works (I've only recently started looking into this).

Cheers
Simon

**Comment by stephenlane** [comment]
Hmmm what version of learning locker are you using. I have been using this exact approach up until v1.3.8 I am on v1.3.10 but haven had time to upgrade to latest. There has been a lot of changes to LL recently so that may be the problem

**Comment by creighton** [comment]
That error is just saying that there were no states with that ID. That doesn't necessarily mean a failure. With the wrapper, it first tries to find any existing attempts on the current activity. It does this by asking the LRS for any attempt-state documents. Its at this point that the LRS will return a list of existing states, or give a 404 saying that there aren't any states. 
see..
https://github.com/adlnet/SCORM-to-xAPI-Wrapper/blob/master/SCORMToXAPIFunctions.js#L669

**Comment by T1ckL35** [comment]
Ah, ok. Looks like I don't yet know enough about the inner workings and how it interfaces.

I did have a look in the db and in the documentapi area I see lots of documents as:
    "documentType" : "state",
    "identId" : "http://adlnet.gov/xapi/profile/scorm/activity-state",
    "activityId" : "****/course_version/package-test",

Is this what should be there? I'm just wondering as I don't see anything relating to 'attempt-state' or similar. Some have a content object with an array of attempts (activityId with an attemptId appended in the url) which appears to be correct.

Oh I'm on LL 1.4.1 but as mentioned it might not be relevant.

Simon

**Comment by T1ckL35** [comment]
I've got to the bottom of this and it isn't a problem with the wrapper.

I've been using some test courses that include the pipwerks API adapter and interfaced with APIWrapper,js. The problem was that the pipwerks API had already called LMSInitialize() and interfacing indirectly with (for example) APIWrapper.js->doLMSInitialize() recalled the API's LMSInitialize() resulting in a circular call killing the browser and stopping me debugging more effectively. I have now extracted the xapi calls from APIWrapper.js into our own adapter and things are working as expected.
The learning locker exception has still occurred in the logs but it (as I wrongly presumed and you correctly stated) doesn't stop execution.

Thanks for clarifying things and your quick response, it is greatly appreciated :)

Cheers
Simon
